src/Bicep.Decompiler/TemplateConverter.cs

```diff
@@ -1388,13 +1403,16 @@ private SyntaxBase ParseModule(IReadOnlyDictionary<string, string> copyResourceL
             }
             else
             {
-                var pathProperty = TemplateHelpers.GetNestedProperty(resource, "properties", "templateLink", "uri") ??
-                    TemplateHelpers.GetNestedProperty(resource, "properties", "templateLink", "relativePath");
+                var pathProperty = TemplateHelpers.GetNestedProperty(resource, "properties", "templateLink", "uri") 
+                                ?? TemplateHelpers.GetNestedProperty(resource, "properties", "templateLink", "relativePath");
+                                
+                var idProperty = TemplateHelpers.GetNestedProperty(resource, "properties", "templateLink", "id");
 
-                if (pathProperty?.Value<string>() is not string templatePathString)
+                if ((pathProperty?.Value<string>() is not string  && idProperty?.Value<string>() is not string ))
                 {
-                    throw new ConversionFailedException($"Unable to find \"uri\" or \"relativePath\" properties under {resource["name"]}.properties.templateLink for linked template.", resource);
+                    throw new ConversionFailedException($"Unable to find \"uri\" or \"relativePath\" or \"id\" properties under {resource["name"]}.properties.templateLink for linked template.", resource);
                 }
+                var templatePathString = idProperty?.Value<string>() ?? pathProperty?.Value<string>() ?? "";
```

Comment:
I'd recommend restructuring this a bit (pulling your logic from line 1424 up), to cut down on the casts, usage of `?`, improving variable naming, and avoiding conflating uses of the `templatePathString` property:

```csharp
SyntaxBase modulePath;
Uri? jsonTemplateUri = null;
if (pathProperty?.Value<string>() is string templatePathString)
{
    modulePath = ...
}
else if (idProperty?.Value<string>() is string templateSpecIdString)
{
    modulePath = GetModuleFromId(templateSpecIdString, resource);
}
else
{
    throw new ConversionFailedException...
}
```